### DISCUSSION TITLE
Error when extending system classes/structs like SwiftUI Color

### DISCUSSION THREAD

**Discussion by chbeer** [skiptools/skip#123] [General]
Dear Skip Team,

thank you for this awesome project. I am trying to migrate an existing project to Skip and I have the following problem:

We have an extension to `Color` to make it `Codable`. If this is a good idea is another story.

This leads to the error: "This extension cannot be merged into its extended Kotlin type, because its type is defined outside of this module. Therefore it cannot add additional protocols"

Is there a workaround?

**Comment by chbeer** [comment]
Also happens to other extensions, like extending String

**Reply by aabewhite** [reply]
Yes, there is no way in Kotlin to add protocols via extension. Kotlin only has extension computed properties and extension functions.  We only allow you to add protocols when your extension is in the same module as the extended type, because in that case we can just move the extension code into the type definition during transpilation. 

There is no transparent workaround that I can think of. I don't know the details of your use case, so it's hard to suggest what to do. I haven't run into many cases where something *has* to be an extension, it's just not as clean sometimes when you can't extend (e.g. your stored property could hold the Color as a custom RGB struct that's Codable, then have a computed property that exposes it as a Color).

As an aside: Skip also doesn't yet support getting RGB values via Color.Resolved, so if you're using that, we'd either have to implement it, or maybe doing everything with RGB internally will be the best way for you after all!
